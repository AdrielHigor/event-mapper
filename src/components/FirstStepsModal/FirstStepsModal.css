.modalHeader, .firstSteps {
  display: flex;
  flex-direction: column;
  align-items: center;
}

p {
  font-size: 14px;
}

.description {
  text-align: center;
}

.subTitle {
  font-size: 14px;
  font-weight: bold;
}

.exitButton {
  border: 0;
  border-radius: 5px;
  width: 160px;
  height: 40px;
  background-color: #00FF00;
  font-weight: bold;
  font-size: 14px;
}

.exitButton:hover {
  background-color: #02b102;
  cursor: pointer;
}