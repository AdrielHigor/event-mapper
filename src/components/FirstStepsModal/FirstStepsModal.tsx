import React from "react";
import "./FirstStepsModal.css";
import ModalContainer from "../ModalContainer/ModalContainer";
import { Styles } from "react-modal";

interface IFirstStepsModal {
  modalIsOpen: boolean;
  closeModal: (
    event: React.MouseEvent<Element, MouseEvent> | React.KeyboardEvent<Element>
  ) => void;
  customStyles?: Styles;
}

const FirstStepsModal = ({
  modalIsOpen,
  closeModal,
  customStyles,
}: IFirstStepsModal) => (
  <ModalContainer
    modalIsOpen={modalIsOpen}
    closeModal={closeModal}
    customStyles={customStyles}
  >
    <div className="firstSteps">
      <div className="modalHeader">
        <h2>Bem Vindo Ao Mapa! 🗺️</h2>
        <p className="description">
          📌Aqui você pode visualizar e cadastrar eventos em sua cidade ou no
          mundo!
        </p>
      </div>
      <div>
        <p className="description">Como Faço Isso? É fácil!</p>
        <p className="subTitle">Navegar</p>
        <p>1 - Navegue pelo mapa e poderá ver eventos próximos de você.</p>
        <p>2 - basta clicar em um marcador para ver os detalhes dos eventos.</p>
        <p className="subTitle">Criar Evento</p>
        <p>1 - Selecione uma localidade do seu gosto e cadastre seu evento;</p>
        <p>2 - Basta clicar no mapa e um popup com formulário irá aparecer;</p>
        <p>3 - Agora basta você preencher o formulário e postar os eventos.</p>
      </div>
      <button className="exitButton" onClick={closeModal}>
        {" "}
        Entendido!{" "}
      </button>
    </div>
  </ModalContainer>
);

export default FirstStepsModal;
