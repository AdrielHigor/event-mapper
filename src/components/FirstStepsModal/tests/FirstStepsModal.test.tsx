import React from "react";
import renderer from "react-test-renderer";
import FirstStepsModal, { IFirstStepsModal } from "../FirstStepsModal";

describe("FirstStepsModal", () => {
  const defaultProps: IFirstStepsModal = {
    modalIsOpen: true,
    closeModal: jest.fn(),
    customClassName: "test-class",
  };

  const wrapper = (options = defaultProps) => {
    return renderer.create(<FirstStepsModal {...options} />);
  };

  it("should render correctly with default props", () => {
    const component = wrapper();
    expect(component).toMatchSnapshot();
  });

  it("should call the closeModal function when the button is clicked", () => {
    const closeModalMock = jest.fn();
    const component = wrapper({ ...defaultProps, closeModal: closeModalMock });

    const exitButton = component.root.findByProps({
      className: "exit-button",
    });

    exitButton.props.onClick();

    expect(closeModalMock).toHaveBeenCalled();
  });
});
