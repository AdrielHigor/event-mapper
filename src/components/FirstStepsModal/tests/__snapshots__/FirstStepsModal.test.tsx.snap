// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstStepsModal should render correctly with default props 1`] = `
<div>
  <div
    className="first-steps"
  >
    <div
      className="modal-header"
    >
      <h2>
        Bem Vindo Ao Mapa! 🗺️
      </h2>
      <p
        className="description"
      >
        📌Aqui você pode visualizar e cadastrar eventos em sua cidade ou no mundo!
      </p>
    </div>
    <div>
      <p
        className="description"
      >
        Como Faço Isso? É fácil!
      </p>
      <p
        className="sub-title"
      >
        Navegar
      </p>
      <p>
        1 - Navegue pelo mapa e poderá ver eventos próximos de você.
      </p>
      <p>
        2 - basta clicar em um marcador para ver os detalhes dos eventos.
      </p>
      <p
        className="sub-title"
      >
        Criar Evento
      </p>
      <p>
        1 - Selecione uma localidade do seu gosto e cadastre seu evento;
      </p>
      <p>
        2 - Basta clicar no mapa e um popup com formulário irá aparecer;
      </p>
      <p>
        3 - Agora basta você preencher o formulário e postar os eventos.
      </p>
    </div>
    <button
      className="exit-button"
      onClick={[MockFunction]}
    >
       
      Entendido!
       
    </button>
  </div>
</div>
`;
